src/MuiComponent.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import DefaultRawTheme from './styles/raw-themes/light-raw-theme';
+import ThemeManager from './styles/theme-manager';
+
+export default class MuiComponent extends React.Component {
+
+  static DEFAULT_THEME;
+
+  static contextTypes = {
+    muiTheme: React.PropTypes.object,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+
+    if (!context || !context.muiTheme) {
+      if (!MuiComponent.DEFAULT_THEME) {
+        MuiComponent.DEFAULT_THEME = ThemeManager.getMuiTheme(DefaultRawTheme);
+      }
+      this.state = {muiTheme: MuiComponent.DEFAULT_THEME};
+    } else {
+      this.state = {muiTheme: context.muiTheme};
+    }
+  }
+
+  componentWillReceiveProps(nextProps, nextContext) {
```

Comment:
Will we have a conflict with https://github.com/callemall/material-ui/blob/master/src/snackbar.jsx#L148?